### DISCUSSION TITLE
TUI SCIM Add error: Unhandled exception in event loop:

### DISCUSSION THREAD

**Discussion by BainMcKay** [JanssenProject/jans#7116] [Q&A] [answered]
In TUI, when I try to add a SCIM config, I get the following error. 

`Unhandled exception in event loop:`
 ` File "/opt/jans/jans-cli/plugins/030_scim/main.py", line 90, in coroutine`
  `  self.app_config = response.json()``
 ` File "/usr/lib/python3/dist-packages/requests/models.py", line 888, in json`
`    return complexjson.loads(`
`  File "/usr/lib/python3/dist-packages/simplejson/__init__.py", line 518, in loads`
`    return _default_decoder.decode(s)`
`  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 370, in decode`
`    obj, end = self.raw_decode(s)`
`  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 400, in raw_decode`
`    return self.scan_once(s, idx=_w(s, idx).end())`
`Exception Expecting value: line 1 column 1 (char 0)`

It displays the error on the console, followed by 

`Press ENTER to continue...`

When IU hit return it goes back to TUI with <set Scim configuration> displayed in red. There is no way I can find to enter SCIM config in TUI. I'll try CLI or API (CURL) . But I'd like to get TUI working.

Any ideas?

**Comment by BainMcKay** [comment, marked as answer]
If I run form CLI,


I get

`{`
`  "baseDN": "o=jans",`
`  "applicationUrl": "https://[FQDN]",`
`  "baseEndpoint": "https://[FQDN]"/jans-scim/restv1",`
`  "personCustomObjectClass": "jansCustomPerson",`
`  "oxAuthIssuer": "https://[FQDN]"",`+
`  "maxCount": 200,`
`  "bulkMaxOperations": 30,`
`  "bulkMaxPayloadSize": 3072000,
`  "userExtensionSchemaURI": "urn:ietf:params:scim:schemas:extension:gluu:2.0:User",`
`  "loggingLevel": "INFO",`
`  "loggingLayout": "text",
`  "metricReporterInterval": 300,`
`  "metricReporterKeepDataDays": 15,`
`  "metricReporterEnabled": true,`
`  "disableJdkLogger": true,`
`  "useLocalCache": true`
`}`

Now it works from TUI. So trunning  './config-cli.py --operation-id get-scim-config' seems tio have fixed something.

**Comment by nynymike** [comment]
Maybe @devrimyatar has an idea...

**Reply by BainMcKay** [reply]
I have too many threads in play. I need to shut them down. 

But on Digital Ocean I have it down to this. I cannot validate the device. jans-auth.log is showing an exception. Here are the 2 lines of log. It's authenticating [`Authentication success for Client`] but has an error on the response [`ErrorResponseFactory.java:93) - Error not found, id: authorization_pending`]


`2024-01-14 09:32:33,614 INFO  [qtp1260134048-16] 87fc87fc-b713-439b-a2a4-e89c2a1295fa [io.jans.as.server.auth.Authenticator] (Authenticator.java:301) - Authentication success for Client: '2000.98dbeb5c-cf6d-4962-82fb-09795a5d6bd2'`
`2024-01-14 09:32:33,653 ERROR [qtp1260134048-16] 87fc87fc-b713-439b-a2a4-e89c2a1295fa [io.jans.as.model.error.ErrorResponseFactory] (ErrorResponseFactory.java:93) - Error not found, id: authorization_pending`

I can't find a way to move beyond this.  Can someone provide some guidance?

**Comment by BainMcKay** [comment]
Resolved.

DO not use OpenDJ. No longer supported;

use MySQL
